tests/ad_divgradsimple.cpp

```diff
@@ -0,0 +1,217 @@
+template <typename Scalar>
+Scalar divgradsimple_x(Scalar x1, Scalar y1, Scalar z1);
+template <typename Scalar>
+Scalar divgradsimple_y(Scalar x1, Scalar y1, Scalar z1);
+template <typename Scalar>
+Scalar divgradsimple_z(Scalar x1, Scalar y1, Scalar z1);
+
+typedef ShadowNumber<double, long double> RawScalar;
+const unsigned int NDIM = 3;
+typedef DualNumber<RawScalar, NumberVector<NDIM, RawScalar> > FirstDerivType;
+typedef DualNumber<FirstDerivType, NumberVector<NDIM, FirstDerivType> > SecondDerivType;
+typedef SecondDerivType ADType;
+
+template<typename Scalar>
+int run_test()
+{
+  // CCP: I hacked this threshold to be very lenient of errors
+  const Scalar thresh = 5000 * std::numeric_limits<Scalar>::epsilon();
+  // parameters
+  Scalar x, y, z;
+
+  //problem size
+  int nx = 5;  // number of points
+  int lx=10;     // length
+  Scalar dx=Scalar(lx)/Scalar(nx);
+  Scalar dy=Scalar(lx)/Scalar(nx);
+  Scalar dz=Scalar(lx)/Scalar(nx);
+
+  // errors
+  Scalar uerror, verror, werror;
+
+  // evaluate MMS (1D)
+  for(int i=0;i<nx;i++)
+  {
+    x = i*dx;
+    for (int j=0;j<nx;j++)
+    {
+        y = j*dy;
+        for (int k=0;k<nx;k++)
+        {
+            z = k*dz;
+            Scalar ad_x_soln = divgradsimple_x<Scalar>(x,y,z);
+            Scalar ad_y_soln = divgradsimple_y<Scalar>(x,y,z);
+            Scalar ad_z_soln = divgradsimple_z<Scalar>(x,y,z);
+
+            Scalar exact_x_soln = 6.0*x + 4.0;
+            Scalar exact_y_soln = 6.0*y + 4.0;
+            Scalar exact_z_soln = 6.0*z + 4.0;
+
+	// test the result is roughly zero
+	// choose between abs and rel error
+#ifdef MASA_STRICT_REGRESSION
+
+            uerror = std::abs(ad_x_soln - exact_x_soln);
+            verror = std::abs(ad_y_soln - exact_y_soln);
+            werror = std::abs(ad_z_soln - exact_z_soln);
+
+#else
+
+            uerror = std::abs(ad_x_soln - exact_x_soln)/std::abs(exact_x_soln);
+            verror = std::abs(ad_y_soln - exact_y_soln)/std::abs(exact_y_soln);
+            werror = std::abs(ad_z_soln - exact_z_soln)/std::abs(exact_z_soln);
+
+#endif	
+	        threshcheck(uerror,thresh);
+	        threshcheck(verror,thresh);
+	        threshcheck(werror,thresh);
+        }
+    }
+  }
+
+  // tests passed
+  return 0;
+
+} // end run_test
+
+template <typename Scalar>
+Scalar divgradsimple_x(Scalar x1, Scalar y1, Scalar z1)
+{
+
+  typedef DualNumber<Scalar, NumberVector<NDIM, Scalar> > FirstDerivType;
+  typedef DualNumber<FirstDerivType, NumberVector<NDIM, FirstDerivType> > SecondDerivType;
+  typedef DualNumber<SecondDerivType, NumberVector<NDIM, SecondDerivType> > ThirdDerivType;
+  typedef ThirdDerivType ADScalar;
+
+  //
+  NumberVector<NDIM, ADScalar> U;
+
+  ADScalar x = ADScalar(x1,NumberVectorUnitVector<NDIM, 0, Scalar>::value());
+  ADScalar y = ADScalar(y1,NumberVectorUnitVector<NDIM, 1, Scalar>::value());
+  ADScalar z = ADScalar(z1,NumberVectorUnitVector<NDIM, 2, Scalar>::value());
+
+  // Assign a simple manufactured solution to U
```

Comment:
No, but it's not; this isn't even a PDE solution.  This really is a proper unit test of ```div(grad(U)) == analytic_div_grad_U```, not just a bloated regression test.